app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt

```diff
@@ -0,0 +1,126 @@
+package org.oppia.app.home.topiclist
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.recyclerview.widget.RecyclerView
+import androidx.databinding.library.baseAdapters.BR
+import org.oppia.app.R
+import org.oppia.app.databinding.PromotedStoryCardBinding
+import org.oppia.app.databinding.TopicSummaryViewBinding
+import org.oppia.app.databinding.WelcomeBinding
+import org.oppia.app.home.UserAppHistoryViewModel
+import org.oppia.app.model.TopicSummary
+
+private const val VIEW_TYPE_WELCOME_MESSAGE = 1
+private const val VIEW_TYPE_PROMOTED_STORY = 2
+private const val VIEW_TYPE_TOPIC_LIST = 3
+
+/** Adapter to inflate different items/views inside [RecyclerView]. The itemList consists of various ViewModels. */
+class TopicListAdapter(
+  private val itemList: MutableList<Any>,
```

Comment:
Let's avoid ``Any``. I suggest introducing an abstract ``HomeItemViewModel`` supertype for each of the three view models contained in this adapter, and using downcasting as necessary to convert to more specific types. This provides better type safety than ``Any``, and makes it possible to convert this adapter to a generic bindable adapter in the future. See #270 in StateFragmentPresenter how that adapter was set up with a common supertype, and similarly in #195.

Reply:
Introduced `HomeItemViewModel` and made all viewmodels extend `HomeItemViewModel`.